pkg/controller/servicebinding/servicebinder.go

```diff
@@ -298,7 +298,37 @@ func (b *serviceBinder) bind() (reconcile.Result, error) {
 
 	b.logger.Debug("Saving data on intermediary secret...")
 
-	secretObj, err := b.secret.createOrUpdate(b.envVars, b.sbr.AsOwnerReference())
+	if b.sbr.Status.Secret != "" {
```

Comment:
Never rely on something that has been written previously: the name should be stable enough for us to use only it forward on.

Instead of checking the `b.sbr.Status.Secret` field, build the name using the `HashedResourceNameForData` function and always update it at the end of the reconciliation loop.

Also, it seems this block is pretty similar to `createOrUpdate` internals.

Reply:
>  build the name using the HashedResourceNameForData function and always update it at the end of the reconciliation loop 

I don't have the secret data at this point to build the name. I use the name to fetch the content and check it against the incoming payload.

Reply:
> Also, it seems this block is pretty similar to createOrUpdate internals.

This block compares the secret reference in ServiceBinding and incoming payload

`createOrUpdate` checks if the secret with the same name already exists.. and then compares its content

Reply:
> > build the name using the HashedResourceNameForData function and always update it at the end of the reconciliation loop
> 
> I don't have the secret data at this point to build the name. I use the name to fetch the content and check it against the incoming payload.

The whole point of your change was to let the content be known in `newSecret`, so the contents are known when the name is known.

`createOrUpdate` should not rely on reading the secret to decide whether it should be updated or not since the update won't (or should not) go through if the payload doesn't change anyways later in the process.

Reply:
Done

Reply:
This piece of code is removed